anomdetect/src/main/java/com/expedia/adaptivealerting/anomdetect/breakout/Edmx.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2018-2019 Expedia Group, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expedia.adaptivealerting.anomdetect.breakout;
+
+import lombok.experimental.UtilityClass;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+
+import java.util.Arrays;
+
+/**
+ * Breakout detector (a.k.a. change point detector) based on the E-Divisive with Exact Medians (EDM-X) algorithm as
+ * described in "Leveraging Cloud Data to Mitigate User Experience from Breaking Bad", by James et al. See
+ * https://arxiv.org/abs/1411.7955 for details.
+ */
+@UtilityClass
+@Slf4j
+public class Edmx {
+
+    /**
+     * Runs EDM-X on the given time series data. This method scales the data to [0, 1] per the Appendix.
+     *
+     * @param data  Time series data
+     * @param delta Minimum sample size for computing a median. This applies to both the left and the right medians.
+     * @return EDM-X breakout result
+     */
+    public static BreakoutResult edmx(double[] data, int delta) {
+        val scaledData = unitScale(data);
+        val n = scaledData.length;
+
+        double bestStat = Double.MIN_VALUE;
```

Comment:
Hm, I think this is wrong. With this, we will always have a breakout. Will look into this a bit more.